sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala

```diff
@@ -329,10 +326,17 @@ private[spark] object HiveUtils extends Logging {
 
       val classLoader = Utils.getContextOrSparkClassLoader
       val jars = allJars(classLoader)
-      if (jars.length == 0) {
-        throw new IllegalArgumentException(
-          "Unable to locate hive jars to connect to metastore. " +
-            s"Please set ${HIVE_METASTORE_JARS.key}.")
+      if (SystemUtils.isJavaVersionAtLeast(JavaVersion.JAVA_9)) {
+        // Do nothing. The system classloader is no longer a URLClassLoader in Java 9,
```

Comment:
Do you mean we are unable to detect the missing Hive jar with Java 9+?

Reply:
The code above tries to get the URLs of all the JARs in the classpath from the system classloader. This isn't available in Java 9. However these only seem to be needed in 'builtin' mode, in which case (I _believe_) we don't need to do this at all -- don't make a new ClassLoader, just use the current ClassLoader.

@squito has an alternative, to read the classpath off of `java.class.path`. I think that works too. It's less change, but, I was also wondering whether the whole idea of making a new ClassLoader in this mode isn't necessary anyway.

Reply:
I think your approach makes more sense.  I was a little worried that someone would add a jar directly to classloader, from java code, which wouldn't show up on java.class.path.  probably an unsupported gray area anyway, but this might make it work in a few more cases